src/components/new-safe/create/logic/utils.ts

```diff
@@ -0,0 +1,17 @@
+import { computeNewSafeAddress } from '@/components/new-safe/create/logic/index'
+import { isSmartContract } from '@/hooks/wallets/web3'
+import type { DeploySafeProps } from '@safe-global/protocol-kit'
+import type { BrowserProvider } from 'ethers'
+
+export const getAvailableSaltNonce = async (provider: BrowserProvider, props: DeploySafeProps): Promise<string> => {
+  const safeAddress = await computeNewSafeAddress(provider, props)
+  const isContractDeployed = await isSmartContract(provider, safeAddress)
```

Comment:
Could we not run into Rate Limit issues here? If someone deployed e.g. 10-20 Safes already?

Reply:
I haven't had any issues so far with around 10 safes created but we should handle possible error responses from the RPC call. In case this becomes an issue with users we could still optimize this e.g. by not incrementally checking saltNonces but using something like binary search.

Reply:
I don't think this will realistically happen like what is the use-case for a wallet to create the exact same safe setup 30 times? I agree though that we should try to limit test it cc. @francovenica

Reply:
We could also add a fallback to the readonly provider in case the wallet provider fails to increase the chances of success.